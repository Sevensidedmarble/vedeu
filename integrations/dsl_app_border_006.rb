#!/usr/bin/env ruby

# frozen_string_literal: true

require 'bundler/setup'
require 'vedeu'

TESTCASE = 'dsl_app_border_006'

class DSLApp

  Vedeu.bind(:_initialize_) { Vedeu.trigger(:_refresh_) }

  Vedeu.configure do
    debug!
    height 10
    log Dir.tmpdir + '/vedeu.log'
    renderers [
                Vedeu::Renderers::Terminal.new(
                  filename: Dir.tmpdir + "/#{TESTCASE}.out",
                  write_file: true),
                # Vedeu::Renderers::JSON.new(filename: Dir.tmpdir + "/#{TESTCASE}.json"),
                # Vedeu::Renderers::HTML.new(filename: Dir.tmpdir + "/#{TESTCASE}.html"),
                # Vedeu::Renderers::Text.new(filename: Dir.tmpdir + "/#{TESTCASE}.txt"),
              ]
    run_once!
    standalone!
  end

  load File.dirname(__FILE__) + '/support/test_interface.rb'

  Vedeu.border :test_interface do
    foreground   '#ff5722'
    top_left     'A', colour: { background: '#ff0000' }
    top_right    'B', colour: { background: '#0000ff', foreground: '#ffffff' }
    bottom_left  'C', colour: { background: '#ffff00', foreground: '#000000' }
    bottom_right 'D', colour: { background: '#ffffff' }
  end

  Vedeu.render do
    view(:test_interface) do
      lines do
        line 'custom'
        line 'vertices'
      end
    end
  end

  def self.start(argv = ARGV)
    Vedeu::Launcher.execute!(argv)
  end

end # DSLApp

Vedeu.timer('Test') do
  DSLApp.start
end

load File.dirname(__FILE__) + '/test_runner.rb'
TestRunner.result(TESTCASE, __FILE__)
