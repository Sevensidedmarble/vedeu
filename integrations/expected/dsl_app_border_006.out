[0m[2;2H[48;5;196m(0A(B[2;3H[38;5;202m[49m(0q(B[2;4H(0q(B[2;5H(0q(B[2;6H(0q(B[2;7H(0q(B[2;8H(0q(B[2;9H(0q(B[2;10H(0q(B[2;11H(0q(B[2;12H(0q(B[2;13H(0q(B[2;14H(0q(B[2;15H(0q(B[2;16H(0q(B[2;17H(0q(B[2;18H(0q(B[2;19H(0q(B[2;20H(0q(B[2;21H(0q(B[2;22H(0q(B[2;23H(0q(B[2;24H(0q(B[2;25H(0q(B[2;26H(0q(B[2;27H(0q(B[2;28H(0q(B[2;29H(0q(B[2;30H(0q(B[2;31H(0q(B[2;32H[38;5;231m[48;5;21m(0B(B[0m[3;2H[38;5;202m[49m(0x(B[3;3H[39m[49mc[3;4Hu[3;5Hs[3;6Ht[3;7Ho[3;8Hm[3;32H[38;5;202m[49m(0x(B[0m[4;2H[38;5;202m[49m(0x(B[4;3H[39m[49mv[4;4He[4;5Hr[4;6Ht[4;7Hi[4;8Hc[4;9He[4;10Hs[4;32H[38;5;202m[49m(0x(B[0m[5;2H[38;5;202m[49m(0x(B[5;32H(0x(B[0m[6;2H[38;5;16m[48;5;226m(0C(B[6;3H[38;5;202m[49m(0q(B[6;4H(0q(B[6;5H(0q(B[6;6H(0q(B[6;7H(0q(B[6;8H(0q(B[6;9H(0q(B[6;10H(0q(B[6;11H(0q(B[6;12H(0q(B[6;13H(0q(B[6;14H(0q(B[6;15H(0q(B[6;16H(0q(B[6;17H(0q(B[6;18H(0q(B[6;19H(0q(B[6;20H(0q(B[6;21H(0q(B[6;22H(0q(B[6;23H(0q(B[6;24H(0q(B[6;25H(0q(B[6;26H(0q(B[6;27H(0q(B[6;28H(0q(B[6;29H(0q(B[6;30H(0q(B[6;31H(0q(B[6;32H[48;5;231m(0D(B[0m